class7.1/src/MaybeTheBestCodeEver.cpp

```diff
@@ -0,0 +1,112 @@
+#include <iostream>
+#include <vector>
+#include <algorithm>
+#include <limits> 
+#include <cstddef> 
+
+class Rectangle {
+public:
+    Rectangle(double width, double height) : width(width), height(height), area(width * height) {}
+
+    double getWidth() const { return width; }
+    double getHeight() const { return height; }
+    double getArea() const { return area; }
+
+    bool canFitInside(const Rectangle& other) const {
+        return width <= other.getWidth() && height <= other.getHeight();
+    }
+
+private:
+    double width;
+    double height;
+    double area;
+};
+
+class RectangleManager {
+public:
+    void addRectangle(const Rectangle& rectangle) {
+        rectangles.push_back(rectangle);
+        updateBiggestArea(rectangle.getArea());
+        updateSmallestArea(rectangle.getArea());
+    }
+
+    const std::vector<Rectangle>& getRectangles() const {
```

Comment:
minor: it may be reasonable to mark this method as nodiscard: if someone had requested reference to a big object and ignored return then something maybe wrong with the client. nodiscard allows to catch such type of logical errors